integr8ly/grafana-operator
# INTLY-10438 - Implement conditional serviceaccount creation

## Description

- Implements a new field in the Grafana CRD to allow users to skip ServiceAccount creation, this means that the service account will have to be created manually and won't be reconciled by the operator
- Serviceaccount creation can be skipped with `Spec.skipCreateServiceAccount: True`

## Relevant issues/tickets

https://issues.redhat.com/browse/INTLY-10438

## Type of change




## Checklist


## Verification steps

***These steps verify scenarios where the user will switch from a managed serviceaccount to an unmanaged one and back to a managed account***
- Deploy Grafana as normal
- Run the operator
- Ensure the serviceaccount is created
- Reapply the Grafana CR with `Spec.skipCreateServiceAccount: True`
- Ensure the operator no longer reconciles the `grafana-serviceaccount` serviceaccount with `kubectl get serviceaccounts -n <your namespace>` 
- `grafana-serviceaccount` should no longer be recreated
- Remove `Spec.skipCreateServiceAccount: True` or switch it to `False`
- Ensure the operator recreates the serviceaccount
